src/Xenial.Tasty/Reporters/SummaryReporter.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xenial.Delicious.Metadata;
+using Xenial.Delicious.Utils;
+
+namespace Xenial.Delicious.Reporters
+{
+    public static class SummaryReporter
+    {
+        private static readonly List<TestCase> cases = new List<TestCase>();
+
+        public static void Register()
+            => Tasty.RegisterReporter(Report)
+                    .RegisterSummaryProvider(Summary);
+
+        public static Task Report(TestCase test)
+        {
+            cases.Add(test);
+            return Task.CompletedTask;
+        }
+
+        public static Task<TestSummary> Summary()
+        {
+            return Task.FromResult(new TestSummary(cases));
+        }
+    }
+
+    public class TestSummary : IReadOnlyList<TestCase>
+    {
+        private readonly List<TestCase> tests;
+        private readonly Dictionary<TestOutcome, List<TestCase>> testsPerOutCome;
+        private readonly Dictionary<TestOutcome, TestResult> result;
+
+        public TestSummary(IEnumerable<TestCase> tests)
```

Comment:
I am not 100% happy with putting so much logic into the `ctor`, we should move that to a static factory method and do a normal value assign public ctor